snmp/datadog_checks/snmp/snmp.py

```diff
@@ -356,7 +351,16 @@ def _start_discovery(self):
                 host_config = self._build_config(instance)
                 self._config.discovered_instances[host] = host_config
 
-        self._thread = threading.Thread(target=self.discover_instances, name=self.name)
+        raw_discovery_interval = self._config.instance.get('discovery_interval', 3600)
+        try:
+            discovery_interval = float(raw_discovery_interval)
+        except (ValueError, TypeError):
+            message = 'discovery_interval could not be parsed as a number: {!r}'.format(raw_discovery_interval)
+            raise ConfigurationError(message)
+
+        self._thread = threading.Thread(
+            target=functools.partial(self.discover_instances, interval=discovery_interval), name=self.name
```

Comment:
You can just pass `args` instead of using partial.

Reply:
Sure, nothing against it.